src/client/api/tests/job_tests.c

```diff
@@ -73,7 +73,7 @@ int uname(struct utsname *buf)
 		return -1;
 	}
 
-	strncpy(buf->nodename, uname_nodename, _UTSNAME_LENGTH);
+	strncpy(buf->nodename, uname_nodename, _UTSNAME_LENGTH - 1);
```

Comment:
This assumes that nodename has a null byte at _UTSNAME_LENGTH - 1.  It should be followed with buf->nodename[_UTSNAME_LENGTH - 1] = '\0'; 

to be safe.

Reply:
Good catch. Fixing.

Reply:
Actually going to memset the entire buffer to ensure zeroes after the string copied.